libimage/filters.go

```diff
@@ -158,33 +158,54 @@ func (r *Runtime) compileImageFilters(ctx context.Context, options *ListImagesOp
 	return filterFuncs, nil
 }
 
+func filterNames(filter string, names, registries []string) (bool, error) {
```

Comment:
The entire function should be replaced with https://github.com/containers/image/blob/main/docker/reference/helpers.go#L36.  That will match the exact behavior of Docker (I tracked it down in the moby code).

Reply:
I don't believe that is true.  We need to remove the registry just like docker remove docker.io to be our own flavor.  But if you have other ideas, I say take this PR over and make it the way you believe it should be.

Reply:
Here's the location in moby: https://github.com/moby/moby/blob/master/daemon/images/images.go#L145